code/modules/power/lighting.dm

```diff
@@ -93,31 +115,33 @@
 		if(W.item_flags & NODROP)
 			to_chat(user, "<span class='warning'>[W] is stuck to your hand!</span>")
 			return
-		user.dropItemToGround(W)
 		if(cell)
-			user.visible_message("<span class='notice'>[user] swaps [W] out for [src]'s cell.</span>", \
-			"<span class='notice'>You swap [src]'s power cells.</span>")
-			cell.forceMove(drop_location())
-			user.put_in_hands(cell)
+			to_chat(user, "<span class='warning'>There is a power cell already installed!</span>")
+			return
 		else
 			user.visible_message("<span class='notice'>[user] hooks up [W] to [src].</span>", \
 			"<span class='notice'>You add [W] to [src].</span>")
-		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
-		W.forceMove(src)
-		cell = W
-		add_fingerprint(user)
-		return
+			playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+			user.dropItemToGround(W)
```

Comment:
```suggestion
			if (!user.temporarilyRemoveItemFromInventory(W))
						//add handling if the user can't drop the cell here
						return
```
You should probably move the visible message and sound below this too

Reply:
From what I could see, every other instance of this ifcheck simply returns, so I did that here as well.